projects/spectator/src/lib/matchers.ts

```diff
@@ -240,18 +240,42 @@ export const toBeEmpty = comparator(el => {
 });
 
 /**
- * The :hidden selector selects hidden elements.
- * Hidden elements are elements that are:
- * 1. Set to display:none
+ * Hidden elements are elements that have:
+ * 1. Display property set to "none"
  * 2. Width and height set to 0
  * 3. A hidden parent element (this also hides child elements)
- * 4. Form elements with type="hidden"
+ * 4. Type equal to "hidden" (only for form elements)
+ * 5. A "hidden" attribute
+ */
+function isHidden(el) {
+  while (el) {
```

Comment:
Why is the check not only on the element?

Reply:
You mean, why did I abstracted the isHidden function instead of writing it directly inside the matcher function?
Because it's used both for `toBeVisible` and `toBeHidden` matcher.

Otherwise I didn't understood the question, could you rephrase it?

Reply:
I mean, why traversed the tree? The check should be only on the element itself.

Reply:
@NetanelBasal In the old JSDoc it says

```
 * Hidden elements are elements that are:
...
 * 3. A hidden parent element (this also hides child elements)
```

Which is probably taken from https://api.jquery.com/hidden-selector/

So the previous jQuery implementation was probably using this methodology too.

Reply:
Yes, I know. I meant, do we really need it in a spec context? Usually I want to test the exact element.

Reply:
Dunno, but changing it will probably need a major version bump.

Reply:
Yes, let's leave it like this.